tests/scripts/validateSparkFvtOutput.py

```diff
@@ -0,0 +1,231 @@
+	else:
+		print("Actual and Expected results differed for the evaluated Patient Cohort for: ", act_to_exp_cnt, " evaluations.")
+		failing+=1
+		xmlfile.write("\t\t"+ "<testcase classname=\"sparkValidation\" name=\"validatePatientCohortEval\"><failure message=\"Actual and Expected results differed for the evaluated Patient Cohort for: "+ str(act_to_exp_cnt) +" evaluations.\">fail</failure></testcase>" + "\n")
+		
+#Function to validate results for the Practitioner Cohort
+def validatePractitionerCohortEval():
+	global passing
+	global failing
+	global xmlfile
+	
+	actual_df = spark.read.parquet("spark-cos/fvt-output/Practitioner_cohort")
+	
+	expected_data = [
+           Row(1, False),
+           Row(2, False),
+           Row(3, True),
+		   Row(4, True),
+		   Row(5, True),
+		   Row(6, True),
+		   Row(7, False),
+		   Row(8, True),
+		   Row(9, True),
+		   Row(10, False)
+	]
+
+	sch = StructType([StructField("id", IntegerType(), False),\
+          StructField("PractitionerMeasureFVT|cohort", BooleanType(), True)])
+
+	expected_df = spark.createDataFrame(expected_data, sch)
+	print(" ")
+	print("Expected Practitioner_Cohort Results:")
+	expected_df.show()
+	print("------------------------------------------------")
+	print(" ")
+	print("Actual Practitioner_Cohort Results read in Parquet format from Cloud Object Storage:")
+	actual_df.sort(["id"]).show()
+
+    #Do the validation of actual results to expected results
+	exp_to_act_cnt=expected_df.exceptAll(actual_df.select(["id", "PractitionerMeasureFVT|cohort"])).count()
+	act_to_exp_cnt=actual_df.exceptAll(expected_df.select(["id", "PractitionerMeasureFVT|cohort"])).count()
+
+	if((exp_to_act_cnt == 0) and (act_to_exp_cnt == 0)):
+		print("Actual results match the Expected results for the evaluated Practitioner Cohort.")
+		passing+=1
+		xmlfile.write("\t\t"+ "<testcase classname=\"sparkValidation\" name=\"validatePractitionerCohortEval\"/>" + "\n")
+	else:
+		print("Actual and Expected results differed for the evaluated Practitioner Cohort for: ", act_to_exp_cnt, " evaluations.")
+		failing+=1
+		xmlfile.write("\t\t"+ "<testcase classname=\"sparkValidation\" name=\"validatePractitionerCohortEval\"><failure message=\"Actual and Expected results differed for the evaluated Practitioner Cohort for: "+ str(act_to_exp_cnt) +" evaluations.\">fail</failure></testcase>" + "\n")
+
+#main() function that calls each of the validation functions.
+def main():
+	global passing
+	global failing
+	global xmlfile
+
+	# write out header portion of xml file
+	xmlfile.write("<?xml version='1.0' encoding='UTF-8'?>" +"\n")
+	xmlfile.write("<testsuites>" + "\n")
+	xmlfile.write("\t"+ "<testsuite name=\"Spark Fvt Tests\">" + "\n")
+	# call individual validation functions in order
+	validateDeviceCohortEval()
+	validateObservationCohortEval()
+	validatePatientCohortEval()
+	validatePractitionerCohortEval()
```

Comment:
There appears to be a lot of duplicated code in these four methods. If you passed in the filename and expectedData, would this be able to be a single function instead of four separate methods?